src/components/Avatar/Avatar.vue

```diff
@@ -0,0 +1,233 @@
+<!--
+  - @copyright Copyright (c) 2018 Julius Härtl <jus@bitgrid.net>
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div v-tooltip="displayName" v-click-outside="closeMenu"
+		:class="{ 'icon-loading': loading, 'unknown': unknown }"
+		:style="avatarStyle"
+		class="avatardiv popovermenu-wrapper" @click="toggleMenu">
+		<img v-if="!loading && !unknown" :src="avatarUrlLoaded">
+		<div v-if="unknown" class="unknown">{{ initials }}</div>
+		<div v-show="openedMenu" class="popovermenu">
+			<popover-menu :is-open="openedMenu" :menu="menu" />
+		</div>
+	</div>
+</template>
+
+<script>
+
+/* global OC oc_userconfig */
+
+import { VTooltip } from 'v-tooltip'
+import { PopoverMenu } from '../PopoverMenu'
+import ClickOutside from 'vue-click-outside'
+import axios from 'nextcloud-axios'
+import uidToColor from './uidToColor'
+
+export default {
+	name: 'Avatar',
+	directives: {
+		tooltip: VTooltip,
+		ClickOutside: ClickOutside
+	},
+	components: {
+		PopoverMenu
+	},
+	props: {
+		/**
+		 * Set a custom url to the avatar image
+		 */
+		url: {
+			type: String,
+			default: undefined
+		},
+		/**
+		 * Set the user id to fetch the avatar
+		 */
+		user: {
+			type: String,
+			default: undefined
```

Comment:
No required props?
I'm guessing there must be at least a minimal setup here? :)

Reply:
I couldn't find a way to define that either user or url must be defined. Any hints how that could be achieved are very welcome.

Reply:
You can just use required: true. https://github.com/nextcloud/vue-components/blob/master/src/components/PopoverMenu/PopoverMenuItem.vue#L77
That way it will throw an error if not using the bare minimal? :woman_shrugging: